### ISSUE TITLE
[Bug] Allow zero replica for Helm

### ISSUE THREAD

**Issue by ducviet00** [ray-project/kuberay#965]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

Others

### What happened + What you expected to happen

I'm using helm for creating ray cluster. I modified replica number and minReplica number in helm values but it doesn't work. 

In the helm values file:
```yaml
additionalWorkerGroups:
  smallGroup:
    replicas: 0
    minReplicas: 0
    maxReplicas: 3
```

When CRDs deployed:
```yaml
- groupName: smallGroup
    replicas: 1
    minReplicas: 1
    maxReplicas: 3
```

Expected:
```yaml
- groupName: smallGroup
    replicas: 0
    minReplicas: 0
    maxReplicas: 3
```

### Reproduction script

Set the worker replica number to zero in values file
```yaml
# Default values for ray-cluster.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

# The KubeRay community welcomes PRs to expose additional configuration
# in this Helm chart.

image:
  repository: rayproject/ray
  tag: 2.0.0
  pullPolicy: IfNotPresent

nameOverride: "kuberay"
fullnameOverride: ""

imagePullSecrets: []

head:
  labels: {}
  rayStartParams:
    dashboard-host: '0.0.0.0'
    block: 'true'
  containerEnv: []
  envFrom: []
  resources:
    limits:
      cpu: "1"
      memory: "2G"
    requests:
      cpu: "1"
      memory: "2G"
  annotations: {}
  nodeSelector: {}
  tolerations: []
  affinity: {}
  securityContext: {}
  volumes:
    - name: log-volume
      emptyDir: {}
  volumeMounts:
    - mountPath: /tmp/ray
      name: log-volume
  sidecarContainers: []
  command: []
  args: []


worker:
  groupName: workergroup
  replicas: 1
  labels: {}
  rayStartParams:
    block: 'true'
  initContainerImage: 'busybox:1.28'  # Enable users to specify the image for init container. Users can pull the busybox image from their private repositories.
  initContainerSecurityContext: {}
  containerEnv: []
  envFrom: []
  resources:
    limits:
      cpu: "1"
      memory: "1G"
    requests:
      cpu: "1"
      memory: "1G"
  annotations: {}
  nodeSelector: {}
  tolerations: []
  affinity: {}
  securityContext: {}
  volumes:
    - name: log-volume
      emptyDir: {}
  volumeMounts:
    - mountPath: /tmp/ray
      name: log-volume
  sidecarContainers: []
  command: []
  args: []

additionalWorkerGroups:
  smallGroup:
    # Disabled by default
    disabled: true
    replicas: 0
    minReplicas: 0
    maxReplicas: 3
    labels: {}
    rayStartParams:
      block: 'true'
    initContainerImage: 'busybox:1.28'  # Enable users to specify the image for init container. Users can pull the busybox image from their private repositories..
    initContainerSecurityContext: {}
    containerEnv: []
    envFrom: []
    resources:
      limits:
        cpu: 1
        memory: "1G"
      requests:
        cpu: 1
        memory: "1G"
    annotations: {}
    nodeSelector: {}
    tolerations: []
    affinity: {}
    securityContext: {}
    volumes:
      - name: log-volume
        emptyDir: {}
    volumeMounts:
      - mountPath: /tmp/ray
        name: log-volume
    sidecarContainers: [].
    command: []
    args: []

service:
  type: ClusterIP
```
Install helm chart.
`helm upgrade --install --create-namespace -n ray-v2 ray-helm .` 

Get the values of CRD.
`kubectl get rayclusters -n ray-v2 ray-helm-kuberay -o yaml`

You can see minReplica number is 1

### Anything else

This happens because ternary function in template file. `{{ 0 | 1 }} = 1`
https://github.com/ray-project/kuberay/blob/87dde227d95457949a22a6cfea41bfbaeca46116/helm-chart/ray-cluster/templates/raycluster-cluster.yaml#L91
https://github.com/ray-project/kuberay/blob/87dde227d95457949a22a6cfea41bfbaeca46116/helm-chart/ray-cluster/templates/raycluster-cluster.yaml#L157

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by ducviet00** [comment]
Created PR: #966

**Comment by nvtkaszpir** [comment]
kuberay raycluster updates are extremely limited:
```
While updating replicas and workersToDeleteUpdate is supported, updating other fields in RayCluster manifests is not 
supported. In particular, updating Ray head pod and Ray worker pod configuration is not supported.
To update pod configuration, delete the RayCluster, edit its configuration and then re-create the cluster
In other words, use kubectl delete and kubectl create to update a RayCluster's pod configuration,
rather than kubectl apply.
Support for in-place updates of pod configuration is tracked in KubeRay issue [#527](https://github.com/ray-project/kuberay/issues/527).

```
https://github.com/ray-project/kuberay/issues/527

**Comment by kevin85421** [comment]
Closed by #968.
